tools/reviewer/aa/aa_tool.sh

```diff
@@ -10,16 +10,45 @@
 # To compile aa from a workspace: 'export AA_FORCE_COMPILE_WS=<>'
 # To undo: 'unset AA_FORCE_COMPILE_WS'
 
-# TODO:
-# 1. Add function set_BAZEL_WORKSPACE(), to set variable to either head or debug ws.
-# 2. Add function bazel_run(force_build) to run a bazel command:
-#    1. cd into workspace
-#    2. Check if file exists
-#    3. If not (or if force_build), bazel build it
-#    4. Exit if build fails
-#    5. Run command
-# 3. Use bazel_run() in _aa_completions() and in aa(), set_BAZEL_WORKSPACE() in aa()
-# 4. Make sure that after this, `aa` can be run from outside of a workspace (but inside base).
+function set_STARTUP_OS_REPO() {
+    set_AA_BASE
+    if [[ -z "$AA_BASE" ]]; then
+    echo "BASE file not found in path until root"
+    return 1
+
+    fi
+    if [[ -z "$AA_FORCE_COMPILE_WS" ]]; then
+        # we want to use `aa` from head
+        export STARTUP_OS_REPO="$AA_BASE/head/startup-os/"
+    else
+        # there's a workspace we want to use `aa` from
+        export STARTUP_OS_REPO="$AA_BASE/ws/$AA_FORCE_COMPILE_WS/startup-os/"
+    fi
+}
+
+function bazel_run() {
+    local target=$1
+    shift;
+    local force_compile=$1
+    shift;
+    local args="$@"
+    local binary_for_target=$(echo "$target" | sed -e 's|//||g; s|:|/|g; s|^|bazel-bin/|g')
+
+    set_STARTUP_OS_REPO
+    pushd $STARTUP_OS_REPO
+
+    if [[ ! -f ${binary_for_target} ]] || [[ "${force_compile}" -eq 1 ]]; then
+        bazel build ${target} &> /dev/null
```

Comment:
We didn't `cd` into `STARTUP_OS_REPO` right?
Does `bazel` run it from there then?

Reply:
@oferb 

>We didn't cd into STARTUP_OS_REPO right?
we did in line 38 😉 

`pushd $STARTUP_OS_REPO`

Reply:
Right.. ;)